src/api/app/models/review.rb

```diff
@@ -94,23 +97,31 @@ def check_initial
       errors.add(:unknown, 'no reviewer defined')
     end
 
-    if by_user && !User.find_by_login(by_user)
+    if by_user
+      error = true if by_group || by_project || by_package
+    elsif by_group
+      error = true if by_project || by_package
+    end
+
+    errors.add(:base, 'it is not allowed to have more than one reviewer entity: by_user, by_group, by_project, by_package.') if error
+
+    if by_user && !entity
       errors.add(:by_user, "#{by_user} not found")
     end
 
-    if by_group && !Group.find_by_title(by_group)
+    if by_group && !entity
```

Comment:
I know that you only changed on thing here... but better if we use guard clause style